src/software/world/ball_model/ball_model.h

```diff
@@ -0,0 +1,29 @@
+#pragma once
+
+#include <optional>
+
+#include "software/time/duration.h"
+#include "software/world/ball_state.h"
+
+/**
+ * A BallModel predicts the future state of the ball
+ */
+class BallModel
+{
+   public:
+    /**
+     * Returns the estimated state of the ball at the specified amount of time in the
+     * future
+     *
+     * @param second_in_future The Duration into the future at which to predict the
+     * ball's position from when the BallModel was initialized
+     *
+     * @throws std::invalid_argument if the ball is estimating state at a time from the
+     * past
+     *
+     * @return The future state of the ball
+     */
+    virtual BallState estimateFutureState(double seconds_in_future) = 0;
```

Comment:
Should use `Duration` instead of `double` for time-based parameters

Reply:
fixed for all ball model